src/metabase/core.clj

```diff
@@ -63,10 +63,11 @@
   []
   (let [hostname  (or (config/config-str :mb-jetty-host) "localhost")
         port      (config/config-int :mb-jetty-port)
-        setup-url (str "http://"
-                       (or hostname "localhost")
-                       (when-not (= 80 port) (str ":" port))
-                       "/setup/")]
+        site-url  (or (public-settings/site-url)
```

Comment:
`(public-settings/site-url)` ensures the url does not have a trailing slash, whereas `(config/config-str :mb-site-url)` would not.